### PR TITLE
Add hotkeys to diagram

### LINKED ISSUE #105: Cut and Paste feature doesn't work on OSX desktop version 1.1.0
I'm unable to get the Cut and Past feature to work on OSX desktop version 1.1.0 from the "Edit" menu or when using command x (⌘X) and command v ( ⌘V).

### PR DESCRIPTION
**Pull request by bakejeyner**
This is my first PR against the Threat Dragon codebase. Depending on circumstances, I may or may not be able to really hero this PR through the entire review process, but I found it to be valuable to open a PR anyway. Feel free to take inspiration from the changes if you find it best to do so.

I added hotkeys to the diagram to unselect the currently selected cell and to copy and paste a selected cell. The hotkey logic leverages the already existing functionality to do so.

I did not update tests, but they still pass. I had trouble getting the hotkeys object to inject into `diagram-spec.js`, and I'm starting to hit the limit of my timebox, so I felt it most appropriate to throw up a PR rather than let my work die without chance.

Functional changes:
- Pressing Escape while focused on the diagram will unselect the currently selected cell
- Pressing Ctrl-C while focused on the diagram will create a reference to the currenly selected cell
- Pressing Ctrl+V while focused on the diagram will duplicate the copied element

Nonfunctional changes:
- In diagramming.js, make created paper focus on mousedown and add a tabindex of 0 (in order to focus). Also add some css to ignore the outline on focus.
- Added hotkey.js to package.json, and pass in hotkey as a service in /services/index.js
- Added *.iml to .gitignore

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
   "private": false,
   "scripts": {
     "build": "npm run bundle-css && npm run build-templates",
-    "build-templates": "./node_modules/.bin/npm-html2js -b src -i 'src/**/*.html' --module templates -o './src/templates.js'",
+    "build-templates": "./node_modules/.bin/npm-html2js -b src -i \"src/**/*.html\" --module templates -o ./src/templates.js\"",
```
Comment (bakejeyner) [inline package.json]: Single quotes were causing `src/templates.js` to not be generated by the npm build task. Adding escaped double quotes fixed this issue.

File: src/services/diagramming.js [hunk 2]
```diff
@@ -113,6 +113,11 @@ function diagramming() {
             interactive: interactive
         });
 
+        paper.el.setAttribute('tabindex', '0');
+        paper.el.addEventListener('mousedown', function() {
+            this.focus();
+        });
+
```
Comment (bakejeyner) [inline src/services/diagramming.js:120]: Had to add the tabindex to make the paper focusable. That way the hotkeys will only execute when the paper is in focus, and not when doing something else like filling out threat details.

**Comment by jgadsden** [comment]
Many thanks @bakejeyner for this PR, this is a very valuable addition.  I will test it on MacOS and Linux as well. We are planning to release version 1.2 after this weekend, but we might delay the release to get this change in. Thanks again!

File: package.json (hunk 1 above)
Comment (jgadsden) [reply package.json]: Hello @bakejeyner , good point - ti sounds like this was causing problems in IntelliJ IDE? I think there may also be a typo, could you try:

`"build-templates": "./node_modules/.bin/npm-html2js -b src -i \"src/**/*.html\" --module templates -o \"./src/templates.js\"",`

(with the addition of more escaped double quotes)

File: src/diagrams/diagram.js [hunk 3]
```diff
@@ -72,6 +77,13 @@ function diagram($scope, $location, $routeParams, $timeout, dialogs, common, dat
         onSelectElement();
     });
 
+
+    $scope.$on('$destroy', function () {
+        hotkeys.unbind("Escape");
+        hotkeys.unbind("Ctrl+C");
```
Comment (jgadsden) [inline src/diagrams/diagram.js]: for MacOS could we have the following changes:
```
-    hotkeys("Ctrl+C", copyHotkey);
-    hotkeys("Ctrl+V", pasteHotkey);
+    hotkeys("CmdOrCtrl+C", copyHotkey);
+    hotkeys("CmdOrCtrl+V", pasteHotkey);
```

**Review by jgadsden — COMMENTED** [review]
Hello @bakejeyner - this is very good to see. I agree that there is more work to be done, as also on MacOS the copied element does not get pasted back into the diagram.
Thanks again for the PR, and when I have more time I can have another look to see what is needed to get this all working.

File: package.json (hunk 1 above)
Comment (bakejeyner) [reply package.json]: >it sounds like this was causing problems in IntelliJ IDE?

Actually, the problem occurred when running the npm scripts from the command line. For example:
```
npm run build
```
would not output the `src/templates.js` file.

But surprisingly, running:
```
./node_modules/.bin/npm-html2js -b src -i 'src/**/*.html' --module templates -o './src/templates.js'
```
on the command line would output the `src/templates.js` file.

**Review by jgadsden — APPROVED** [review]
Thanks for the updates @bakejeyner , it works for me now. I think we should merge it into master, and extend the test coverage at a later date  - if you agree?

**Comment by bakejeyner** [comment]
Sounds good to me! Thank you!

**Pull request merged**